tests/contrib/affiliations/test_affiliations_resource.py

```diff
@@ -0,0 +1,153 @@
+):
+    """Example affiliation."""
+    aff = service.create(identity, affiliation_full_data)
+    Affiliations.index.refresh()  # Refresh the index
+
+    return aff
+
+
+def test_affiliations_invalid(client, h, prefix, affiliation_full_data):
+    """Test invalid type."""
+    # invalid type
+    res = client.get("{prefix}/invalid}", headers=h)
+    assert res.status_code == 404
+
+
+def test_affiliations_forbidden(
+    client, h, prefix, example_affiliation, affiliation_full_data
+):
+    """Test invalid type."""
+    # invalid type
+    affiliation_full_data_too = deepcopy(affiliation_full_data)
+    affiliation_full_data_too["id"] = "other"
+    res = client.post(
+        f"{prefix}", headers=h, data=json.dumps(affiliation_full_data_too))
+    assert res.status_code == 403
+
+    res = client.put(
+        f"{prefix}/cern", headers=h, data=json.dumps(affiliation_full_data))
+    assert res.status_code == 403
+
+    res = client.delete(f"{prefix}/cern")
+    assert res.status_code == 403
+
+
+def test_affiliations_get(client, example_affiliation, h, prefix):
+    """Test the endpoint to retrieve a single item."""
+    id_ = example_affiliation.id
+
+    res = client.get(f"{prefix}/{id_}", headers=h)
+    assert res.status_code == 200
+    assert res.json["id"] == id_
+    # Test links
+    assert res.json["links"] == {
+        "self": "https://127.0.0.1:5000/api/affiliations/cern"
+    }
+
+
+def test_affiliations_search(client, example_affiliation, h, prefix):
+    """Test a successful search."""
+    res = client.get(prefix, headers=h)
+
+    assert res.status_code == 200
+    assert res.json["hits"]["total"] == 1
+    assert res.json["sortBy"] == "name"
+
+
+def _create_affiliations(service, identity):
+    """Create dummy affiliations with similar names/acronyms/titles."""
+    affiliations = [
+        {
+            "acronym": "CERN",
+            "id": "cern",
+            "name": "CERN",
+            "title": {
+                "en": "European Organization for Nuclear Research",
+                "fr": "Conseil Européen pour la Recherche Nucléaire"
+            }
+        },
+        {
+            "acronym": "OTHER",
+            "id": "other",
+            "name": "OTHER",
+            "title": {
+                "en": "CERN"
+            }
+        },
+        {
+            "acronym": "CERT",
+            "id": "cert",
+            "name": "CERT",
+            "title": {
+                "en": "Computer Emergency Response Team",
+                "fr": "Équipe d'Intervention d'Urgence Informatique"
+            }
+        },
+        {
+            "acronym": "NU",
+            "id": "nu",
+            "name": "Northwestern University",
+            "title": {
+                "en": "Northwestern University",
+            }
+        }
+    ]
+    for aff in affiliations:
+        service.create(identity, aff)
+
+    Affiliations.index.refresh()  # Refresh the index
+
+
+def test_affiliations_suggest_sort(
```

Comment:
I am assuming this is the type of behaviour we want *only* on suggest... wouldn't it be nice on search too?

Reply:
I'm not sure what you mean with the question - perhaps ask me in chat.